central/globaldb/bolt.go

```diff
@@ -25,6 +28,10 @@ func initialize() {
 
 // GetGlobalDB returns a pointer to the global db.
 func GetGlobalDB() *bolt.DB {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		utils.Should(errors.New("unexpected call, should not open BoltDB while using Postgres"))
+	}
+
```

Comment:
This has some overlap with the following call: postgres.LogCallerOnPostgres. It should not be called any more. It could be better if we rename/repurpose it.